shipgen/EvtCalcGenerator.cxx

```diff
@@ -0,0 +1,121 @@
+  if (fTree->FindBranch("vx") ){fTree->SetBranchAddress("vx", &vx);}
+  if (fTree->FindBranch("vy") ){fTree->SetBranchAddress("vy", &vy);}
+  if (fTree->FindBranch("vz") ){fTree->SetBranchAddress("vz", &vz);}
+
+  if (fTree->FindBranch("px_prod1") ){fTree->SetBranchAddress("px_prod1", &px_prod1);}
+  if (fTree->FindBranch("py_prod1") ){fTree->SetBranchAddress("py_prod1", &py_prod1);}
+  if (fTree->FindBranch("pz_prod1") ){fTree->SetBranchAddress("pz_prod1", &pz_prod1);}
+  if (fTree->FindBranch("e_prod1") ){fTree->SetBranchAddress("e_prod1", &e_prod1);}
+  if (fTree->FindBranch("pdg_prod1") ){fTree->SetBranchAddress("pdg_prod1", &pdg_prod1);}
+
+  if (fTree->FindBranch("px_prod2") ){fTree->SetBranchAddress("px_prod2", &px_prod2);}
+  if (fTree->FindBranch("py_prod2") ){fTree->SetBranchAddress("py_prod2", &py_prod2);}
+  if (fTree->FindBranch("pz_prod2") ){fTree->SetBranchAddress("pz_prod2", &pz_prod2);}
+  if (fTree->FindBranch("e_prod2") ){fTree->SetBranchAddress("e_prod2", &e_prod2);}
+  if (fTree->FindBranch("pdg_prod2") ){fTree->SetBranchAddress("pdg_prod2", &pdg_prod2);}
+
+  if (fTree->FindBranch("px_prod3") ){fTree->SetBranchAddress("px_prod3", &px_prod3);}
+  if (fTree->FindBranch("py_prod3") ){fTree->SetBranchAddress("py_prod3", &py_prod3);}
+  if (fTree->FindBranch("pz_prod3") ){fTree->SetBranchAddress("pz_prod3", &pz_prod3);}
+  if (fTree->FindBranch("e_prod3") ){fTree->SetBranchAddress("e_prod3", &e_prod3);}
+  if (fTree->FindBranch("pdg_prod3") ){fTree->SetBranchAddress("pdg_prod3", &pdg_prod3);}
+
+  return kTRUE;
+}
+// -------------------------------------------------------------------------
+
+
+// -----   Destructor   ----------------------------------------------------
+EvtCalcGenerator::~EvtCalcGenerator()
+{
+ // cout << "destroy Ntuple" << endl;
+ fInputFile->Close();
+ fInputFile->Delete();
+ delete fInputFile;
+}
+// -------------------------------------------------------------------------
+
+// -----   Passing the event   ---------------------------------------------
+Bool_t EvtCalcGenerator::ReadEvent(FairPrimaryGenerator* cpg)
+{
+  if (fn==fNevents){ LOG(WARNING) << "End of input file. Rewind."; }
+  fTree->GetEntry(fn);
+  fn++;
+  if (fn %100==0)  cout << "Info EvtCalcGenerator: event nr "<< fn << endl;
+
+  //Vertex coordinates in the SHiP reference frame, expressed in [cm]
+  Double_t space_unit_conv = 100.;            // m to cm 
+  Double_t coord_shift     = (zDecayVolume - ztarget)/space_unit_conv; // units m
+  Double_t vx_transf = vx * space_unit_conv; // units cm
+  Double_t vy_transf = vy * space_unit_conv; // units cm
+  Double_t vz_transf = (vz - coord_shift) * space_unit_conv; // units cm
+
+  Double_t c = 2.99792458e+10; // speed of light [cm/s]
+  Double_t tof = TMath::Sqrt(vx_transf*vx_transf + vy_transf*vy_transf 
+                            + vz_transf*vz_transf)/c;
+  
+// Mother LLP
+  Bool_t wanttracking=false;
+  pdg_llp = 0.; // Geantino, placeholder
```

Comment:
Bad idea to use 0 here, which is by the way not the geantino (should be 998/999).

Better to follow the [standard numbering scheme](https://pdg.lbl.gov/2024/reviews/rpp2024-rev-monte-carlo-numbering.pdf) (see Section 11) or use the placeholder 99.

Reply:
Implemented. I have left the Geantino placeholder for now. In the future it would be ideal to set up appropriate PDG codes within EvtCalc for each of the LLP.